SpringProject/src/main/java/com/gradproject2019/users/controller/UserController.java

```diff
@@ -19,6 +20,12 @@
     @Autowired
     private UserService userService;
 
+    @GetMapping(path = "/users/search")
```

Comment:
I'm not saying _don't_ do this - but if you look at https://internal-docs.staging.corp.mendeley.com/apidocs/docs (appreciate some things in their aren't perfect) see how these implement 'searching' e.g. profiles-service

Most have the base `/users/` path with optional query params to allow _filtering_ which strictly speaking is what you are doing. This is because `list` endpoints can and should be filterable/sortable on many columns - but this will make your code for the list dao more complex. 

For now, I'm a believer in making things work first, so I've got no problem with this in a sandbox project - but be aware its not the most ideal thing to do

Reply:
I agree with your point regarding the filtering, and I think it's something we might implement after the feature freeze when we're addressing our technical debt

Reply:
sure - i didn't mean do it now - more just something to think about